### PR TITLE
Comprehensive improvements

### PR DESCRIPTION
**Pull request by SerafimArts**
Added features:
 - Basic options for Activity execution
 - Query and Signal handlers support
 - Basic PSR-3 implementation
 - Separated workflow executor
 - Complete generation of attribute-based declarations

Fixed bugs:
 - Incorrect answer when resolving a promise
 - Memory leaks when resolving deferreds

Well and something else...

### CODE REVIEW

File: example/Workflow/PizzaDelivery.php [hunk 1]
```diff
@@ -11,39 +11,40 @@
...
+        $b = yield $context->executeActivity('B');
 
-        return 42;
+        return [$a, $b];
+    }
+
+    /**
+     * @QueryMethod()
+     */
+    #[QueryMethod]
```
Comment (wolfy-j) [inline example/Workflow/PizzaDelivery.php:36]: is it 8 ready?

File: example/client.php [hunk 2]
```diff
@@ -29,18 +32,15 @@
 //
 $worker = new Worker($transport, $loop);
 
+$worker->setLogger($logger);
```
Comment (wolfy-j) [inline example/client.php:35]: we will need per workflow logger as well, with replaying capabilities

**Review by wolfy-j — APPROVED** [review]
Crazy amount of work!

File: example/Workflow/PizzaDelivery.php (hunk 1 above)
Comment (SerafimArts) [reply example/Workflow/PizzaDelivery.php:36]: Yep. You can choose one annotations type. 

Here it is duplicated in different syntax for the example.

File: example/client.php (hunk 2 above)
Comment (SerafimArts) [reply example/client.php:35]: It should be located somewhere here: https://github.com/temporalio/php-sdk/blob/develop/src/Runtime/Workflow/Executor.php#L79

**Pull request merged**